src/clarity-icons/clarity-icons-api.ts

```diff
@@ -88,12 +98,26 @@ export class ClarityIconsApi {
             throw new TypeError("Only string argument is allowed in this method.");
         }
 
-        // if shapeName doesn't exist in the icons templates, throw an error.
-        if (!this.has(shapeName)) {
-            throw new Error(`'${shapeName}' is not found in the Clarity Icons set.`);
+        return iconShapeSources[shapeName];
+    }
+
+    // This method will save which icon is used as well as the reference to its wrapper element, which is clr-icon.
+    // By saving reference to clr-icon, we could access it without DOM traversal and update the injected template
+    // if the icon's template is updated in the API.
+    _cacheAndRetrieve(iconElement: ClarityIconElement, shapeName: string, cachedIconName?: string) {
+        if (!cachedIconName) {
+            cachedIconName = shapeName;
+        }
+
+        if (!usedIconsTemplateCache[cachedIconName]) {
+            usedIconsTemplateCache[cachedIconName] = [iconElement];
```

Comment:
Why are you caching the icon as an array like this? You're only storing it once, and if an override icon is set this cache wouldn't be updated right?

Reply:
I think the term "cache" is more misleading here. I am thinking about renaming it. It collects only the references to the icon instances. It doesn't save any data state. But I am working on a part where if the element is removed from the DOM, the instance gets removed from the cache.